### ISSUE TITLE
[ADB over TCP] Granting package permissions crashes (missing GRANT_RUNTIME_PERMISSIONS permission)

### ISSUE THREAD

**Issue by bb010g** [MuntashirAkon/AppManager#284]
**Describe the bug**
In ADB over TCP mode, `android.permission.IPermissionManager$Stub$Proxy.grantRuntimePermission` is called when granting permissions despite a lack of `android.permission.GRANT_RUNTIME_PERMISSIONS`. This crashes App Manager.

**To Reproduce**
Steps to reproduce the behaviour:
1. Ensure lack of the `android.permission.GRANT_RUNTIME_PERMISSIONS` permission.
2. Ensure App Manager is in ADB over TCP mode.
3. Open permissions tab for an app, like [MatLog](https://github.com/pluscubed/matlog).
4. Flip the grant/deny toggle for a permissions without a related app op permission, like `android.permission.READ_LOGS`.
5. App Manager crashes.

**Expected behavior**
Either:
1. App Manager does not show toggles for permissions it can not grant/deny without root / the `android.permission.GRANT_RUNTIME_PERMISSIONS` permission in a package's permission tab in ADB over TCP mode, as it would in no root mode.
  * I would expect App Manager to be able to toggle the permission over ADB, but I would expect a lack of crashing more.
2. App Manager utilizes ADB to grant/deny permissions in ADB over TCP mode.
  * I would expect this behavior the most.

**Crash logs**
```
java.lang.SecurityException: grantRuntimePermission: Neither user 10360 nor current process has android.permission.GRANT_RUNTIME_PERMISSIONS.
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2373)
    at android.os.Parcel.createException(Parcel.java:2357)
    at android.os.Parcel.readException(Parcel.java:2340)
    at android.os.Parcel.readException(Parcel.java:2282)
    at android.permission.IPermissionManager$Stub$Proxy.grantRuntimePermission(IPermissionManager.java:1351)
    at io.github.muntashirakon.AppManager.servermanager.PackageManagerCompat.grantPermission(PackageManagerCompat.java:108)
    at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.setPermission(AppDetailsViewModel.java:378)
    at io.github.muntashirakon.AppManager.details.AppDetailsFragment$AppDetailsRecyclerAdapter.lambda$null$22$AppDetailsFragment$AppDetailsRecyclerAdapter(AppDetailsFragment.java:1348)
    at io.github.muntashirakon.AppManager.details.-$$Lambda$AppDetailsFragment$AppDetailsRecyclerAdapter$ebSZkn7bHyfqKqD18v0XtWPMWr4.run(Unknown Source:10)
    at java.lang.Thread.run(Thread.java:923)
 Caused by: android.os.RemoteException: Remote stack trace:
 at android.app.ContextImpl.enforce(ContextImpl.java:2018)
 at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2046)
 at com.android.server.pm.permission.PermissionManagerService.grantRuntimePermissionInternal(PermissionManagerService.java:1423)
 at com.android.server.pm.permission.PermissionManagerService.grantRuntimePermission(PermissionManagerService.java:1404)
 at android.permission.IPermissionManager$Stub.onTransact(IPermissionManager.java:648)


Device Info:
App version: 2.5.23-PRE
App version code: 381
Android build version: 7038034
Android release version: 11
Android SDK version: 30
Android build ID: RQ1A.210205.004
Device brand: google
Device manufacturer: Google
Device name: redfin
Device model: Pixel 5
Device product name: redfin
Device hardware name: redfin
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: adb
```

**Device info**
 - Device: Google Pixel 5
 - OS Version: Android 11
 - App Manager Version: 2.5.23-PRE
 - Mode: ADB over TCP

**Additional context**
All root-dependent (permission-dependent) code should probably be gated by default, or at least refactored to behind interfaces that check for relevant permissions first and can leave a note to the user that they'd need to grant that permission to continue that specific task? (Or have a general way to catch & friendlily handle these sort of permission exceptions.)

**Comment by MuntashirAkon** [comment]
This is unrelated to the permission itself. The crash meant that, for some reason, remote service isn't working or pinging remote service failed and it fell back to local service which caused this crash.

**Comment by MuntashirAkon** [comment]
Oops, sorry. I accidentally closed the issue.

**Comment by MuntashirAkon** [comment]
All issues related to ADB should be fixed in a8b24937.
